Tl15

rcgee opened the pull request:
Implements the Tusa and Langer (2015) GMPE for the volcanic area of Mt Etna. The authors consider the functional forms of Sabetta and Pugliese, 1987 (SP87), and a simplified version of Boore and Atkinson, 2008 (BA08). For both functional forms they differentiate between shallow events (SE) and deep events (DE), for a total of 4 GMPEs. Another GMPE has been added that is an unpublished version that uses hypocentral distance, for a total of 5 GMPEs:

1)TusaLanger2015RepiBA08SE
2)TusaLanger2015RepiBA08DE
3)TusaLanger2015RepiSP87SE
4)TusaLanger2015RepiSP87DE
5)TusaLanger2015Rhypo

Note: Due to limited data, authors do not derive coefficients for site class C (and also do not provide guidelines how to handle this case), so an error is raised if the user tries to specify site class C.

All GMPEs have been tested against an excel implementation supplied by authors. Tests are green: https://ci.openquake.org/job/zdevel_oq-hazardlib/405/

g-weatherill commented on openquake/hazardlib/gsim/tusa_langer_2015.py:
```diff
@@ -0,0 +1,375 @@
+        # intensity measure type 
+        C = self.COEFFS[imt]
+
+	imean = (self._compute_magnitude(rup, C) +
```
I think there is an indentation issue here

g-weatherill commented on openquake/hazardlib/gsim/tusa_langer_2015.py:
```diff
@@ -0,0 +1,375 @@
+        """
+        Return standard deviations as defined in tables below
+        """
+       	assert all(stddev_type in self.DEFINED_FOR_STANDARD_DEVIATION_TYPES
```
More indentation issues. Did you run pep8 on the file?

g-weatherill commented on openquake/hazardlib/gsim/tusa_langer_2015.py:
```diff
@@ -0,0 +1,375 @@
+                (C['b2'] * (rup.mag) ** 2)
+
+    def _get_site_amplification(self, sites, C):
+	
```
This blank line is unnecessary

g-weatherill commented on openquake/hazardlib/gsim/tusa_langer_2015.py:
```diff
@@ -0,0 +1,375 @@
+        regression analysis, and ei are dummy variables (0 or 1) used 
+        to denote the different EC8 site classes.
+	"""
+
```
As above

g-weatherill commented on openquake/hazardlib/gsim/tusa_langer_2015.py:
```diff
@@ -0,0 +1,375 @@
+        
+		
+	# Class D; Vs30 < 180 m/s.
+	idx = (sites.vs30 >= 1E-10) & (sites.vs30 < 180.0) 
```
I don't think you need the two logical commands here, just running sites.v30 < 180.0 should be enough.

g-weatherill commented on openquake/hazardlib/gsim/tusa_langer_2015.py:
```diff
@@ -0,0 +1,375 @@
+
+	for value in sites.vs30:
+    		if 180 <= value < 360:
+        		raise Exception('GMPE does not consider class C sites (Vs30 = 180-360 m/s)')
```
These seem to be over-indented.

g-weatherill commented:
Apart from the minor indentation issues it looks good to me (LGTM)

rcgee commented:
Hi Graeme,  I made all the changes you mentioned above, and changed all lines to ensure a maximum of 80 characters, and improved the formatting of the coefficient tables. The file now passes the command line PEP8 test (with the exception of the length of the coefficient tables of course).

Your suggestion to convert all indentation tabs to spaces has fixed the issue of the indentations appearing different in my fork compared to here, in the PR.

g-weatherill commented:
Great, that seems to have fixed the indentation issues. One last thing, can you update the changelog to note that you have added this new feature: https://github.com/gem/oq-hazardlib/blob/master/debian/changelog

Something like this to the top of the changelog file:
``[Robin Gee]``
``* Adds Tusa and Langer (2015) GMPE``
is fine

daniviga commented:
I also opened a PR to add @rcgee to the contributors list (see #381)

micheles commented on openquake/hazardlib/tests/gsim/tusa_langer_2015_test.py:
```diff
@@ -0,0 +1,112 @@
+STDDEV_DISCREP = 0.1
+
+
+class TusaLanger2015RepiBA08SEtestcase(BaseGSIMTestCase):
```
Please rename TusaLanger2015RepiBA08SEtestcase -> TusaLanger2015RepiBA08SETestCase; idem for the other TestCase below

rcgee commented:
Ok, I updated the changelog file and fixed my TestCase class names, as per Micheles. Thank you for the suggestions and for adding me to the contributors list.

micheles commented:
LGTM

rcgee commented:
Thank you Micheles. How should I proceed next?

micheles commented:
Usually after the LGTMs are given the pull requests on hazardlib are merged by the hazard team, so ask Marco or Graeme to merge.
